### PR TITLE
Use `babel-plugin-inline-view-configs` to increase verification of viewmanagers

### PR DESCRIPTION
**Pull request by acoates-ms**
`babel-plugin-inline-view-configs` is a plugin that core uses which among other things adds some runtime validation that the property types and event types being declared in the view configs of viewmangers match the view specs in js.

This PR copies the `babel-plugin-inline-view-configs` plugin into RNW since its not yet publicly published.  Then it applies https://github.com/facebook/react-native/pull/29230 on top of it which is required before the plugin would work outside of Facebooks haste repo.  It also makes some changes to use a local copy of `react-native-codegen` that we have, which is required until `react-native-codegen` is republished after this PR was merged: https://github.com/facebook/react-native/pull/28827

Finally we apply this plugin in the playground app, so RNTester will hit errors if a viewmanager declares an event or property with the wrong type/parameters.

@TheSavior , one thing to note is that the number of viewmanagers that this currently checks is pretty limited.  It checks `View` and a few of the less important viewmanagers (ex: `Slider`), but there doesn't appear to be a native view spec for `Text`? It also doesnt check `ScrollView` since the view spec for that goes through a redirect in `ScrollViewNativeComponent.js` which only adds the call to `registerGeneratedViewConfig` when running in `RN$Bridgeless` mode.  It also doesn't check `TextInput`, since we forked the native component there too much, so we are not using the native spec.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5431)

### CODE REVIEW

File: vnext/src/overrides.json [hunk 1]
```diff
@@ -3,6 +3,30 @@
     "tsconfig.json"
   ],
   "overrides": [
+    {
+      "type": "patch",
```
Comment (NickGerleman) [inline vnext/src/overrides.json:7]: FYI this whole file is about to move with a PR I have out right now

File: packages/playground/babel.config.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
   presets: ['module:metro-react-native-babel-preset'],
+  plugins: [require('../../vnext/src/babel-plugin-inline-view-configs')],
```
Comment (NickGerleman) [inline packages/playground/babel.config.js:3]: Consider adding this to e2etest as well

File: vnext/src/overrides.json (hunk 1 above)
Comment (acoates-ms) [reply vnext/src/overrides.json:7]: yeah, I saw.

Comment (acoates-ms) [reply vnext/src/overrides.json:7]: I'll let your change get in first.

Comment (NickGerleman) [reply vnext/src/overrides.json:7]: Just merged it in, so you should be able to merge now.

**Comment by TheSavior** [comment]
@acoates-ms, the core components (View, Text, TextInput, Image, and ScrollView) are handwritten and not using codegen. The theory has been that these are complex and unique enough that handwriting them is okay. 

I think you've raised a really good point though, enforcing that the other platforms are consistent via build failures is important. The decision came from only thinking about Android and iOS. I've raised this with the members of the team working on Fabric. While I hope we change direction on the plan for codegen of those platforms, I wouldn't expect it to come for those components before 2021. 😕

**Pull request merged**